pkg/gardenlet/controller/backupbucket/reconciler.go

```diff
@@ -234,7 +234,7 @@ func (r *Reconciler) reconcileBackupBucket(
 		return reconcile.Result{}, nil
 	}
 
-	if reconciliationSuccessful {
+	if reconciliationSuccessful && extensionBackupBucket.Status.LastOperation.State == gardencorev1beta1.LastOperationStateSucceeded {
```

Comment:
Why do we need this second condition? I think we can drop `reconciliationSuccessful` if you prefer the latter condition.